### ISSUE TITLE
Create new role that allows view of draft and embargoed works' files and metadata, but does not provide direct edit access

### ISSUE THREAD

**Issue by bdezray** [psu-libraries/scholarsphere#1623]
Create a role that has some but not all of the admin privileges. This role should be able to view the files and metadata for draft and embargoed works, but not have direct edit access. This will be useful for DCN curation and the Adaptive Technologies Team.

**Comment by ajkiessl** [comment]
We should make a umg for these roles.  On login, this will get stored in their groups and give them these permissions.  We can manage this similarly to how regular admin permissions work.

**Comment by bdezray** [comment]
@ajkiessl Will a umg work for non-PSU people and could people be given temporary access into that umg?

**Comment by ajkiessl** [comment]
Since umgs and our authentication scheme won't work for DCN curators, we'll handle the DCN solution in #1737 .

**Comment by EricDurante** [comment]
@ajkiessl 

Currently in the code, I see that there are two conditions that have to be met for a user to have admin permissions in ScholarSphere:

1. They must be in the group `umg-up.ul-dev-team`
2. They must have their `admin_enabled` flag set

Do you know why the authorization logic was designed such that both conditions are necessary? I can see why the group assignment alone wouldn't work that well - given that the group information is only refreshed whenever the user signs in (I think), it wouldn't be possible to immediately revoke authorization. So also having the `admin_enabled` flag allows for immediate revocation, but if we need to have and manage that flag anyway, then what's the point of the groups? Why not just the flag? Am I missing things?

**Comment by EricDurante** [comment]
I assume that the reason for caching the group information in the database and only refreshing it at sign-in is that it's too expensive to query it externally for every request?

**Comment by EricDurante** [comment]
Oh, I see - it's not about revoking authorization (although, maybe it should be?):

> commit 9f006efeb523a44c02a03ac05dbafd85c28295f6
> Author: Adam Wead <awead@users.noreply.github.com>
> Date:   Mon Oct 5 12:41:13 2020 -0400
> 
>     Toggle admin privileges
> 
>     Allows a user, who has admin rights available to them, to disable/enable
>     those rights via their profile page. This is useful if an administrator
>     wants to use the application like a standard user, or to verify access
>     controls to standard users are being observed properly.
> 
>     This also updates the layout and formatting of the profile edit page to
>     make it more consistent with the rest of the forms in the application,
>     and ensure that the user does not have to enter an ORCiD id for their
>     profile.

That being the case, I take it that group membership by itself is sufficient for granting the permissions described in this issue? I definitely can't create groups - so what would you like to name this one?

**Comment by ajkiessl** [comment]
@EricDurante Yes, I think group membership should suffice for this.  I can create a group; we can decide a name at standup.  Anything will do for a default value, maybe `unreleased-access-group`?

I don't know all the lore around groups in ScholarSphere – since I wasn't working on it when implemented – but it works for more broader, customizable access controls like giving access to user made groups.  However, this feature doesn't seem to be used much.  I guess the original ScholarSphere team felt it just made sense to manage admin rights via groups too, since it already existed?  The toggle in the database (as Adam mentions in the commit message) is so admins can act like a regular user and toggle back to admin rights whenever they want.
